### PR TITLE
squid: objclass: deprecate cls_cxx_gather

### PR DESCRIPTION
**Pull request by NitzanMordhai**
backport tracker: https://tracker.ceph.com/issues/66319

---

backport of https://github.com/ceph/ceph/pull/55923
parent tracker: https://tracker.ceph.com/issues/64258

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

### CODE REVIEW

File: src/objclass/objclass.h [hunk 1]
```diff
@@ -153,7 +153,7 @@ extern int cls_get_snapset_seq(cls_method_context_t hctx, uint64_t *snap_seq);
 
 /* gather */
 extern int cls_cxx_gather(cls_method_context_t hctx, const std::set<std::string> &src_objs, const std::string& pool,
-			  const char *cls, const char *method, bufferlist& inbl);
+			  const char *cls, const char *method, bufferlist& inbl) __attribute__ ((deprecated));
```
Comment (rzarzynski) [inline src/objclass/objclass.h:156]: Hmm, I don't see the doc part (release notes) on deprecating the method. If it's already in, a mention in the commit msg should be present.

Comment (NitzanMordhai) [reply src/objclass/objclass.h:156]: @rzarzynski fixed it, it was not picked so i added it

Comment (rzarzynski) [reply src/objclass/objclass.h:156]: Oh, sorry. I know why git has skipped this line: it's already present – https://github.com/ceph/ceph/pull/57819/files#diff-e5807f35e3c85ea0c57bba6c6713ace9a5f379edd4ec033f0ac2ac81ccad9972R188. Let's go with a mention in the commit message.

Comment (NitzanMordhai) [reply src/objclass/objclass.h:156]: @rzarzynski are you approving it?

**Review by rzarzynski — APPROVED** [review]

**Comment by Naveenaidu** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/SQUID#httpstrackercephcomissues68447

**Pull request merged**